modules/core/packet-server/keeper/keeper_test.go

```diff
@@ -0,0 +1,25 @@
+package keeper_test
+
+import (
+	commitmenttypes "github.com/cosmos/ibc-go/v9/modules/core/23-commitment/types"
+	"github.com/cosmos/ibc-go/v9/modules/core/packet-server/types"
+)
+
+const testClientID = "tendermint-0"
+
+func (suite *KeeperTestSuite) TestSetCounterparty() {
+	merklePathPrefix := commitmenttypes.NewMerklePath([]byte("ibc"), []byte(""))
+	counterparty := types.Counterparty{
+		ClientId:         testClientID,
+		MerklePathPrefix: merklePathPrefix,
+	}
+	suite.chainA.App.GetIBCKeeper().PacketServerKeeper.SetCounterparty(suite.chainA.GetContext(), testClientID, counterparty)
+
+	retrievedCounterparty, found := suite.chainA.App.GetIBCKeeper().PacketServerKeeper.GetCounterparty(suite.chainA.GetContext(), testClientID)
+	suite.Require().True(found, "GetCounterparty does not return counterparty")
+	suite.Require().Equal(counterparty, retrievedCounterparty, "Counterparty retrieved not equal")
+
+	retrievedCounterparty, found = suite.chainA.App.GetIBCKeeper().PacketServerKeeper.GetCounterparty(suite.chainA.GetContext(), "client-0")
```

Comment:
maybe note this clientid doesn't exist or use ibctesting constant invalid client id (think it exists)